src/pybind/mgr/rbd_support/mirror_snapshot_schedule.py

```diff
@@ -364,6 +365,97 @@ def run(self) -> None:
+        unsynced_images = []
+        slow_status_images = []
+        total_image_count = 0
+
+        for pool_id in self.images:
+            with self.module.rados.open_ioctx2(int(pool_id)) as ioctx:
+                for namespace in self.images[pool_id]:
+                    ioctx.set_namespace(namespace)
+                    try:
+                        statuses = list(rbd.RBD().mirror_image_status_list(ioctx))
+                    except Exception as e:
+                        self.log.error(
+                            "error getting image statuses from {}/{}: {}".format(
+                                pool_id, namespace, e
+                            )
+                        )
+                        continue
+
+                    for image_id in self.images[pool_id][namespace]:
+                        schedule = self.schedules.find(pool_id, namespace, image_id)
+                        if not schedule:
+                            continue
+                        total_image_count += 1
+                        timestamps = get_image_timestamps(statuses, image_id)
+                        if not timestamps:
+                            continue
+
+                        min_interval_minutes = min(
+                            schedule.items, key=lambda x: x[0].minutes
+                        )[0].minutes
+                        next_snapshot_timestamp = self.get_queue_timestamp(
+                            pool_id, namespace, image_id
+                        )
+                        if not next_snapshot_timestamp:
+                            continue
+                        expected_snapshot_timestamp = (
+                            next_snapshot_timestamp.timestamp()
+                            - (min_interval_minutes * 60)
+                        )
+                        unsynced_images += [
+                            image_id
+                            for local_ts, remote_ts in timestamps
+                            if (remote_ts - local_ts) > min_interval_minutes * 60
+                        ]
+                        slow_status_images += [
+                            image_id
+                            for local_ts, remote_ts in timestamps
+                            if (expected_snapshot_timestamp - remote_ts)
+                            > min_interval_minutes * 60 * 5
+                        ]
+
+        health_checks = {}
+
+        unsynced_count = len(unsynced_images)
+        slow_count = len(slow_status_images)
+        if total_image_count > 0 and (unsynced_count / total_image_count) > 0.1:
+            health_checks["unsynced_images"] = {
+                "severity": "warning",
+                "summary": "More than 10% of images are not synced according to their schedule {}".format(
+                    {unsynced_count} / {total_image_count}
+                ),
+                "detail": [
+                    "List of unsynced images: {}".format(", ".join(unsynced_images))
```

Comment:
Dumping a list of image IDs isn't going to be helpful to the user.  If listing the affected images at all, use their image specs instead of IDs and limit the listing in size -- this probably means generating an actual list instead of a concatenated string and abiding by `mon_health_max_detail` config option.  See how `AUTH_INSECURE_GLOBAL_ID_RECLAIM` health check is handled in `src/mon/HealthMonitor.cc` for an example.